integration-tests/bats/rebase.bats

```diff
@@ -0,0 +1,396 @@
+    [[ "$output" =~ "no rebase in progress" ]] || false
+}
+
+@test "rebase: -i flag required" {
+    run dolt rebase b1
+    [ "$status" -eq 1 ]
+    [[ "$output" =~ "non-interactive rebases not currently supported" ]] || false
+}
+
+@test "rebase: bad args" {
+    run dolt rebase -i
+    [ "$status" -eq 1 ]
+    [[ "$output" =~ "not enough args" ]] || false
+
+    run dolt rebase -i main b1
+    [ "$status" -eq 1 ]
+    [[ "$output" =~ "rebase takes at most one positional argument" ]] || false
+
+    run dolt rebase --abrot
+    [ "$status" -eq 1 ]
+    [[ "$output" =~ "error: unknown option \`abrot'" ]] || false
+
+    run dolt rebase -i foo
+    [ "$status" -eq 1 ]
+    [[ "$output" =~ "branch not found: foo" ]] || false
+}
+
+@test "rebase: cannot rebase with dirty working set" {
+    dolt sql -q "INSERT INTO t1 VALUES (2,2);"
+    run dolt rebase -i b1
+    [ "$status" -eq 1 ]
+    [[ "$output" =~ "cannot start a rebase with uncommitted changes" ]] || false
+}
+
+@test "rebase: cannot rebase during active merge" {
+    dolt checkout b1
+    dolt sql -q "INSERT INTO t1 VALUES (1,2);"
+    dolt add t1
+    dolt commit -m "b1 commit 2"
+
+    run dolt merge main
+    [ "$status" -eq 1 ]
+    [[ "$output" =~ "Automatic merge failed" ]] || false
+
+    run dolt rebase -i main
+    [ "$status" -eq 1 ]
+    [[ "$output" =~ "unable to start rebase while a merge is in progress – abort the current merge before proceeding" ]] || false
+}
+
+@test "rebase: rebase working branch already exists" {
+    dolt checkout b1
+    dolt branch dolt_rebase_b1
+
+    run dolt rebase -i main
+    [ "$status" -eq 1 ]
+    [[ "$output" =~ "fatal: A branch named 'dolt_rebase_b1' already exists." ]] || false
+}
+
+@test "rebase: verify custom script" {
+    touch rebaseScript.sh
+    echo "#!/bin/bash" >> rebaseScript.sh
+    echo "mv rebasePlan.txt \$1" >> rebaseScript.sh
+    chmod +x rebaseScript.sh
+    export EDITOR=$PWD/rebaseScript.sh
+    export DOLT_TEST_FORCE_OPEN_EDITOR="1"
+
+    dolt checkout b1
+    run dolt show head
+    [ "$status" -eq 0 ]
+    COMMIT1=${lines[0]:12:32}
+
+    touch rebasePlan.txt
+    echo "pick $COMMIT1 b1 commit 1" >> rebasePlan.txt
+
+    run dolt rebase -i main
+    [ "$status" -eq 0 ]
+    [[ "$output" =~ "Successfully rebased and updated refs/heads/b1" ]] || false
+
+    run dolt log
+    [ "$status" -eq 0 ]
+    [[ "$output" =~ "b1 commit 1" ]] || false
+    [[ "$output" =~ "main commit 2" ]] || false
+}
+
+@test "rebase: basic rebase" {
+    touch rebaseScript.sh
+    echo "#!/bin/bash" >> rebaseScript.sh
+    chmod +x rebaseScript.sh
+    export EDITOR=$PWD/rebaseScript.sh
+    export DOLT_TEST_FORCE_OPEN_EDITOR="1"
+
+    dolt checkout b1
+    run dolt rebase -i main
+    [ "$status" -eq 0 ]
+    [[ "$output" =~ "Successfully rebased and updated refs/heads/b1" ]] || false
+
+    run dolt log
+    [ "$status" -eq 0 ]
+    [[ "$output" =~ "main commit 2" ]] || false
+}
+
+@test "rebase: invalid rebase plan" {
```

Comment:
When the rebase plan is invalid, I think we should clean up the rebase state, so that the user can rerun `dolt rebase` to try again. It looks like currently, the rebase branch is created and is left around, so users would have to delete that branch or switch over to the SQL shell to try and finish the rebase. It would be nice if they could just quickly try again with `dolt rebase`.